src/main.ts

```diff
@@ -32,4 +32,50 @@ export class Minifier {
           errors.join('\n'));
     }
   }
+
+  visit(node: ts.Node) {
+    switch (node.kind) {
+      case ts.SyntaxKind.PropertyAccessExpression: {
+        let output = '';
+        let propAccessExp = <ts.PropertyAccessExpression>node;
+        output += this.visit(propAccessExp.expression);
+        output += '.';
+        // Adds '$mangled' to an identifier to ensure that the identifier is being altered.
+        output += propAccessExp.name.text + '$mangled';
+        return output;
+      }
+
+      // Emit most of the TypeScript sourcefile untouched by using .getText() because only
+      // particular cases of identifiers (ie: identifiers that are a part of property access
+      // expressions, property declarations) are of renaming interest.
+      default: {
+        let text = node.getText();
+        let children = node.getChildren();
+        let output = ''; // output with renaming.
+        // prevEnd is used to keep track of up to what position the text was copied to the new
```

Comment:
I think the useful things to document here are:
- the general idea of the algorithm: recurse for child nodes, directly emit the text sections of `node` that are not part of a child
- what the indices are relative to (always the text of the node?)
- what is the loop invariant (condition that holds before and after every iteration)

Maybe these can be formulated a bit more succinct?